### PR TITLE
Reviewlist- implement write a review and update API server

### PR DESCRIPTION
**Pull request by cdeng1111**
@fernanda1411 @van-hsieh can you review?thanks.

### CODE REVIEW

File: client/src/components/QuestionsAndAnswers/QuestionsAndAnswers.jsx [hunk 1]
```diff
@@ -45,14 +45,14 @@ const QuestionsAndAnswers = ({productId}) => {
   return (
     <div className='questions-and-answers'>
       <h2>Questions & Answers</h2>
-      <SearchQuestions />
+      {/* <SearchQuestions /> */}
```
Comment (fernanda1411) [inline client/src/components/QuestionsAndAnswers/QuestionsAndAnswers.jsx]: Why? :)

File: client/src/components/QuestionsAndAnswers/QuestionsAndAnswers.jsx [hunk 2]
```diff
@@ -45,14 +45,14 @@ const QuestionsAndAnswers = ({productId}) => {
...
       <QuestionsList
         questionsAndAnswers={getQuestionsAndAnswers()}
         onQuestionHelpulButtonClick={onQuestionHelpulButtonClick}
         onAnswerHelpulButtonClick={onAnswerHelpulButtonClick}
         setShowAddAnswerModal={setShowAddAnswerModal}
       />
-      <QuestionsAndAnswersActions
+      {/* <QuestionsAndAnswersActions
```
Comment (fernanda1411) [inline client/src/components/QuestionsAndAnswers/QuestionsAndAnswers.jsx]: Why? :)

File: client/src/components/Overview/Overview.jsx [hunk 3]
```diff
@@ -41,19 +41,41 @@ class Overview extends React.Component {
     }
   }
 
+  // getProductInfo() {
```
Comment (fernanda1411) [inline client/src/components/Overview/Overview.jsx]: Please remove comment code :)

I see that you did some work on Van's component. Did you sync with her?

File: client/src/components/Overview/Overview.jsx [hunk 4]
```diff
@@ -41,19 +41,41 @@ class Overview extends React.Component {
...
+      .then(response => {
+        console.log('product response data from overview:', response.data);
+        this.setState({
         product: response.data
-      }));
+      })
+    });
   }
 
+  // getStylesInfo() {
```
Comment (fernanda1411) [inline client/src/components/Overview/Overview.jsx]: same as above

File: client/src/components/Overview/Overview.jsx [hunk 5]
```diff
@@ -41,19 +41,41 @@ class Overview extends React.Component {
...
+
   getProductInfo() {
     axios.get(`/products/${this.props.productId}`)
-      .then(response => this.setState({
+      .then(response => {
+        console.log('product response data from overview:', response.data);
+        this.setState({
         product: response.data
-      }));
+      })
```
Comment (fernanda1411) [inline client/src/components/Overview/Overview.jsx]: missing semicolon

File: client/src/components/Overview/Overview.jsx [hunk 6]
```diff
@@ -41,19 +41,41 @@ class Overview extends React.Component {
...
     axios.get(`/products/${this.props.productId}/styles`)
-      .then(response => this.setState({
+      .then(response => {
+        console.log('styles response data from overview:', response.data);
+
+        this.setState({
         styles: response.data.results,
         currStyle: response.data.results[0]
-      }));
+      })
```
Comment (fernanda1411) [inline client/src/components/Overview/Overview.jsx]: semicolon

**Review by fernanda1411 — CHANGES_REQUESTED** [review]
Good Work!!! 
I left some comments and I see that you form is return an error.
image

File: client/src/components/QuestionsAndAnswers/QuestionsAndAnswers.jsx (hunk 1 above)
Comment (cdeng1111) [reply client/src/components/QuestionsAndAnswers/QuestionsAndAnswers.jsx]: reverted.

File: client/src/components/QuestionsAndAnswers/QuestionsAndAnswers.jsx (hunk 2 above)
Comment (cdeng1111) [reply client/src/components/QuestionsAndAnswers/QuestionsAndAnswers.jsx]: reverted.

File: client/src/components/Overview/Overview.jsx (hunk 3 above)
Comment (cdeng1111) [reply client/src/components/Overview/Overview.jsx]: reverted.

File: client/src/components/Overview/Overview.jsx (hunk 5 above)
Comment (cdeng1111) [reply client/src/components/Overview/Overview.jsx]: reverted.

File: client/src/components/Overview/Overview.jsx (hunk 6 above)
Comment (cdeng1111) [reply client/src/components/Overview/Overview.jsx]: reverted.

File: client/src/components/Overview/Overview.jsx (hunk 4 above)
Comment (cdeng1111) [reply client/src/components/Overview/Overview.jsx]: reverted.

**Review by fernanda1411 — APPROVED** [review]
Nice work Chun! :)

**Pull request merged**